mlir/include/mlir/Dialect/Linalg/IR/LinalgStructuredOps.td

```diff
@@ -82,6 +82,11 @@ def GenericOp : LinalgStructuredBase_Op<"generic", [
         types:
           parallel, reduction, window
 
+    Any values within the output operands that are not unindexed by the 
```

Comment:
This shouldnt happen. In Linalg ops the shape of the iteration space is computed from the data. So by construction, there cannot be an output operand that is not unindexed.

Reply:
I can't understand the double negation in "not unindexed".

Otherwise, nothing currently precludes a "diagonal" `(d0, d1) -> (d0, d0)` indexing map for the result. The iteration dimension associated with `d1` can be inferred from another operand/result so the iteration space is fully defined. So while the iteration space is indeed computed from data, it doesn't necessarily imply that the index space of each individual operand/result is fully covered by the mapping from the iteration space (which should be noted somewhere as a requirement for results but not operands).

Reply:
Sorry, I didn't write that clearly - but yes I was trying to capture the condition that @ftynse mentioned above ^

PR updated